applications/nrf5340_audio/src/bluetooth/bt_renderer/volume/bt_rend_vol.c

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "bt_rend_vol_internal.h"
+
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/audio/vcp.h>
+
+#include "macros_common.h"
+#include "bt_rend.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_rend_vol, CONFIG_BT_REND_VOL_LOG_LEVEL);
+
+#define VOLUME_DEFAULT 195
+#define VOLUME_STEP    16
+
+static struct bt_vcp_vol_ctlr *vcs_client_peer[CONFIG_BT_MAX_CONN];
+
+/**
+ * @brief	Get the index of the first available vcs_client_peer.
+ *
+ * @retval	Index if success.
+ * @retval	-ENOMEM if no available indexes.
+ */
+static int vcs_client_peer_index_free_get(void)
+{
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		if (vcs_client_peer[i] == NULL) {
+			return i;
+		}
+	}
+
+	LOG_WRN("No more indexes for VCS peer clients");
+
+	return -ENOMEM;
+}
+
+/**
+ * @brief	Check if the given @p conn has a vcs_client_peer pointer.
+ *
+ * @param[in]	conn	The connection pointer to be checked.
+ *
+ * @retval	True if vcs_client_peer exists.
+ * @retval	False otherwise.
+ */
+static bool vcs_client_peer_exists(struct bt_conn *conn)
+{
+	int ret;
+
+	struct bt_conn *result_conn = NULL;
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		ret = bt_vcp_vol_ctlr_conn_get(vcs_client_peer[i], &result_conn);
+
+		if (!ret && conn == result_conn) {
+			return true;
+		}
+
+		if (ret == -ENOTCONN) {
+			/* VCS client no longer connected, free the index */
+			vcs_client_peer[i] = NULL;
+			return false;
+		}
+	}
+
+	return false;
+}
+
+/**
+ * @brief	Callback handler for the volume state.
+ *
+ * @note	This callback handler will be triggered if volume state has changed,
+ *		or the playback was muted or unmuted.
+ */
+static void vcs_state_ctlr_cb_handler(struct bt_vcp_vol_ctlr *vcs, int err, uint8_t volume,
+				      uint8_t mute)
+{
+	int ret;
+
+	if (err) {
+		LOG_ERR("VCS state callback error: %d", err);
+		return;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		if (vcs == vcs_client_peer[i]) {
```

Comment:
Lots of indentation. Should we return in this check?

Reply:
We can't do a return, but we can do a continue

Reply:
Same for the one below